baw_client_rest/baw_client_rest.py

```diff
@@ -0,0 +1,130 @@
+"""Contains a simple client for BAW REST Endpoints"""
+
+import time
+from os import getenv
+from botocore.exceptions import ClientError
+import boto3
+import requests
+from requests import RequestException
+
+
+class Client:
+    """
+    Class to simplify connecting to the REST endpoints of BAW,
+    Manages retrieving and caching CSRF tokens,
+    and provides a simple method for sending messages to BAW: send_request
+
+    The following env vars must be set:
+    BAW_ENDPOINT - the endpoint to which you wish to send a message,
+    CSRF_CACHE - the name of the DYnamoDB table used as a csrf cache,
+    BAW_CSRF_URL - the BAW endpoint for retrieving CSRF tokens
+
+    To instantiate provide a BAW username and Password, and a logger.
+
+    All logs will be sent to the provided logger
+    """
+
+    CSRF_CACHE = None
+
+    def __init__(self, username, password, logger):
+        self.username = username
+        self.password = password
+        self.logger = logger
+
+        self.url = self._getenv("BAW_ENDPOINT")
```

Comment:
I would suggest that these values be passed in as per `username`, `password`, etc.  rather than being read from environment variables. As they are a hard dependency, it's best to inject them. Not everyone might be using environment variables, or the same environment variable names, etc.